provider/src/test/groovy/au/com/dius/pact/provider/ProviderVerifierSpec.groovy

```diff
@@ -759,4 +759,79 @@ class ProviderVerifierSpec extends Specification {
     result instanceof VerificationResult.Failed
     result.pending == true
   }
+
+  def 'verifyResponseByFactory is able to successfully verify an AsynchronousMessage with MessageAndMetadata'() {
+    given:
+    verifier.responseFactory = { new MessageAndMetadata('{}'.bytes, [:]) }
+    ProviderInfo provider = new ProviderInfo('Test Provider')
+    def failures = [:]
+    Interaction interaction = new Message('verifyResponseByFactory Test Message', [],
+            OptionalBody.body('{}'.bytes, ContentType.JSON), new MatchingRulesImpl(), new Generators(), [:], 'abc123')
+    IConsumerInfo consumer = Stub()
+    def interactionMessage = 'Test'
+
+    when:
+    def result = verifier.verifyResponseByFactory(
+            provider,
+            consumer,
+            interaction,
+            interactionMessage,
+            failures,
+            false
+    )
+
+    then:
+    result instanceof VerificationResult.Ok
+  }
+
+  def 'verifyResponseByFactory is able to successfully verify a SynchronousRequestResponse'() {
+    given:
+    verifier.responseFactory = { ['statusCode': 200, 'headers': [:], 'data': null] }
+    ProviderInfo provider = new ProviderInfo('Test Provider')
+    def failures = [:]
+    Interaction interaction = new RequestResponseInteraction('Test Interaction',
+            [new ProviderState('Test State')], new Request(), new Response(), '12345678')
+    IConsumerInfo consumer = Stub()
+    def interactionMessage = 'Test'
+
+    when:
+    def result = verifier.verifyResponseByFactory(
+            provider,
+            consumer,
+            interaction,
+            interactionMessage,
+            failures,
+            false
+    )
+
+    then:
+    result instanceof VerificationResult.Ok
+  }
+
+  def 'verifyResponseByFactory returns an error result if the factory method fails with an exception'() {
+    given:
+    ProviderInfo provider = new ProviderInfo('Test Provider')
```

Comment:
Is this test passing because the factory function is not set? 
If so, it would be better to be explicit and have it setup a function that throws an exception on invocation

Reply:
that is correct. I will make it a more explicit test